src/main/java/Joey.java

```diff
@@ -0,0 +1,137 @@
+                } catch (JoeyException e) {
+                    System.out.println("OOPS!!! " + e.getMessage());
+                    System.out.println("____________________________________________________________");
+                }
+            } else if (userCommand.startsWith("deadline")) {
+                try {
+                    String[] parts = userCommand.substring(9).split("/by");
+                    if (parts.length != 2) {
+                        throw new JoeyException("Hey girlie, please provide both description and deadline for the deadline task. Follow this format instead: deadline return book /by Sunday");
+                    }
+
+                    String description = parts[0].trim();
+                    String by = parts[1].trim();
+
+                    if (description.isEmpty() || by.isEmpty()) {
+                        throw new JoeyException("Hey girlie, the description and deadline for the deadline task can't be empty. Follow this format instead: deadline return book /by Sunday");
+                    }
+
+                    tasks[taskCount] = new Deadline(description, by);
+                    taskCount++;
+                    System.out.println(" Got it. I've added this task:");
+                    System.out.println("   " + tasks[taskCount - 1]);
+                    System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                    System.out.println("____________________________________________________________");
+                } catch (StringIndexOutOfBoundsException e) {
+                    System.out.println("OOPS!!! Hey girlie, please provide both description and deadline for the deadline task. Follow this format instead: deadline return book /by Sunday");
+                    System.out.println("____________________________________________________________");
+                } catch (JoeyException e) {
+                    System.out.println("OOPS!!! " + e.getMessage());
+                    System.out.println("____________________________________________________________");
+                }
+            } else if (userCommand.startsWith("event")) {
+                try {
+                    String[] parts = userCommand.split("/from");
+                    if (parts.length != 2) {
+                        throw new JoeyException("Hey girlie, please provide both description, start time, and end time for the event task. Follow this format instead: event project meeting /from Mon 2pm /to 4pm");
+                    }
+
+                    String description = parts[0].substring(6).trim();
+                    String[] timingParts = parts[1].split("/to");
+                    if (timingParts.length != 2) {
+                        throw new JoeyException("Hey girlie, please provide both description, start time, and end time for the event task. Follow this format instead: event project meeting /from Mon 2pm /to 4pm");
```

Comment:
This code line exceeds the number of characters allowed per line. Please read the guidelines and follow.